test/conference.js

```diff
@@ -225,6 +225,122 @@ contract('Conference', function(accounts) {
     })
   })
 
+  describe('finalize using attendee bitmap', function(){
+    let non_registered = accounts[4];
+    let admin = accounts[5];
+
+    beforeEach(async function(){
+      await conference.register(twitterHandle, {value:deposit, from:non_owner});
+      await conference.register(twitterHandle, {value:deposit, from:accounts[6]});
+      await conference.register(twitterHandle, {value:deposit, from:accounts[7]});
+      await conference.register(twitterHandle, {value:deposit, from:accounts[8]});
+    })
+
+    it('can be called by owner', async function(){
+      await conference.finalize([1], {from:owner});
+
+      await conference.isAttended(non_owner).should.eventually.eq(true)
+      await conference.attended().should.eventually.eq(1)
+    })
+
+    it('cannot be called by admin', async function(){
+      await conference.grant([admin], {from:owner});
+      await conference.finalize([1], {from:admin}).should.be.rejected;
+    })
+
+    it('cannot be called by non owner', async function(){
+      await conference.finalize([1], {from:non_owner}).should.be.rejected;
+
+      await conference.isAttended(non_owner).should.eventually.eq(false)
+      await conference.attended().should.eventually.eq(0)
+
+      const participant = await conference.participants(non_owner);
+
+      assert.equal(getParticipantDetail(participant, 'attended'), false);
+    })
+
+    it('isAttended is false if attended function for the account is not called', async function(){
+      await conference.isAttended(owner).should.eventually.eq(false)
+    })
+
+    it('cannot be attended twice', async function(){
+      await conference.finalize([1], {from:owner});
+      await conference.finalize([1], {from:owner}).should.be.rejected;
+
+      await conference.isAttended(non_owner).should.eventually.eq(true)
+      await conference.attended().should.eventually.eq(1)
+    })
+
+    it('marks party as ended and enables payout', async function() {
+      await conference.finalize([1], {from:owner});
+
+      await conference.ended().should.eventually.eq(true)
+      await conference.payoutAmount().should.eventually.eq(await conference.payout())
+    })
+
+    it('correctly updates attendee records', async function() {
+      // all attended except accounts[6]
+      // 1 0 1 1
+      // reverse order since we go from right to left in bit parsing:
+      // [ 13 (1101) ]
+
+      await conference.finalize([13], {from:owner});
+
+      await conference.isAttended(non_owner).should.eventually.eq(true)
+      await conference.isAttended(accounts[6]).should.eventually.eq(false)
+      await conference.isAttended(accounts[7]).should.eventually.eq(true)
+      await conference.isAttended(accounts[8]).should.eventually.eq(true)
+      await conference.attended().should.eventually.eq(3)
+    })
+  })
+
+  describe('finalize large party using attendee bitmaps', function(){
```

Comment:
Do we need any testcase for input values out of bound?

Reply:
Already have cases for those.